File: takasugi/Python/NLP-100-08.py
#
# -*- coding: utf-8 -*-


def cipher(x):
    out = ""

    for c in x:
        if (ord('a') - 1 < ord(c) and ord(c) < ord('z') + 1):
            out += chr(219 - ord(c))
        else:
            out += c

    return out

a = "a1b2c3D4"
print cipher(a)

File: takasugi/Ruby/NLP-100-08.rb
#

def cipher(str)
  ary = str.split('')

  ary.map { |c|
    if c =~ /[a-z]/
      (219-c.ord).chr
    else
      c
    end
    }.join
end

encode = cipher('AbCdEfG1Uu')
decode = cipher(encode)

puts encode
puts decode


